ape_solidity/compiler.py

```diff
@@ -666,8 +670,22 @@ def _get_pragma_spec(self, path: Path) -> Optional[NpmSpec]:
 
         compiler_version = pragma_spec.select(self.available_versions)
         if compiler_version:
-            solcx.install_solc(compiler_version, show_progress=False)
+            install_solc(compiler_version, show_progress=False)
         else:
+            # Attempt to use the best-installed version.
+            # NOTE: Use the oldest version available for maximum compatibility.
+            for version in self.installed_versions[::-1]:
```

Comment:
A little update for readability

```suggestion
            for version in reversed(self.installed_versions):
```